src-tauri/src/farmer.rs

```diff
@@ -113,6 +115,20 @@ async fn farm(base_directory: PathBuf, farm_args: FarmingArgs) -> Result<(), any
     }
 
     info!("Connecting to node at {}", node_rpc_url);
+    if let Err(error) = timeout(Duration::from_secs(10), async {
+        loop {
+            if NodeRpcClient::new(&node_rpc_url).await.is_ok() {
+                break;
+            } else {
+                sleep(Duration::from_millis(500)).await;
+            }
+        }
+    })
+    .await
+    {
+        error!("Node is not responding for 10 seconds, farmer is unable to start");
```

Comment:
Is it recoverable from this state though?

Reply:
I don't think so. There should be something wrong with the node in this case. Probably a restart on the app would fix it